src/components/table/column-picker/column-picker.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+        let numberOfSelectedColumns = nativeElement.querySelectorAll('.column-picker-stats')[1];
+        expect(numberOfSelectedColumns.textContent.trim()).toEqual('6 columns added');
+
+        component.selected = [];
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        numberOfSelectedColumns = nativeElement.querySelectorAll('.column-picker-stats')[1];
+        expect(numberOfSelectedColumns.textContent.trim()).toEqual('1 columns added');
+    });
+});
+
+@Component({
+    selector: 'app-column-picker-sort',
+    template: `
+        <ux-column-picker
+            [(selected)]="selected"
+            [(deselected)]="deselected"
+            [groups]="groups"
+            [sort]="sort">
+        </ux-column-picker>
+    `
+})
+export class ColumnPickerSortTestComponent {
+    selected: ReadonlyArray<string | ColumnPickerGroupItem> = [
+        'Type',
+        'Date',
+        'Requested by',
+        'Status',
+        'Completion'
+    ];
+
+    deselected: Array<string | ColumnPickerGroupItem> = [
+        { group: 'Metadata', name: 'Department' },
+        { group: 'Metadata', name: 'Author' },
+        { group: 'Metadata', name: 'Category' },
+        { group: 'Metadata', name: 'Date Created' },
+        { group: 'Metadata', name: 'Date Modified' },
+        'Document ID',
+        'Flag',
+        'From',
+        'Icon',
+        'Importance',
+        'Location',
+        'Location ID',
+        'Message',
+        { group: 'Metadata', name: 'Organization' },
+        'Time',
+        'Time Created',
+        'Time Modified',
+        'Work Completed'
+    ];
+
+    groups: ColumnPickerGroup[] = [
+        { name: 'Metadata', expanded: true }
+    ];
+
+    sort = (a: string | ColumnPickerGroupItem, b: string | ColumnPickerGroupItem) =>
+        (a['name'] ?? a).localeCompare(b['name'] ?? b)
+}
+
+describe('Column Picker Component - Sort Input', () => {
+    let component: ColumnPickerSortTestComponent;
+    let fixture: ComponentFixture<ColumnPickerSortTestComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [TableModule],
+            declarations: [ColumnPickerSortTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(ColumnPickerSortTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should sort grouped columns when the sort input is provided', () => {
+        let names = [];
+        nativeElement.querySelectorAll('.column-picker-tree-node-level-1')
+            .forEach(element => names = [...names, element.textContent]);
+        expect(names).toEqual(names.slice().sort());
+    });
+
+    it('should sort ungrouped columns when the sort input is provided', () => {
```

Comment:
I would suggest checking that the entire column is in the expected state rather than checking grouped and ungrouped separately.
For example:
```ts
expect(deselectedRows[0].innerText).toBe('Metadata', 'row 0');
expect(deselectedRows[1].innerText).toBe('Author', 'row 1');
```

This will be a lot more manageable if you trim the data set down to a minimal number of elements required to assert correct behavior.